eq-author/src/App/section/Design/SectionEditor/SectionIntroduction/index.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { propType } from "graphql-anywhere";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { TransitionGroup } from "react-transition-group";
+
+import AddPage from "App/QuestionnaireDesignPage/icon-add-page.svg?inline";
+import MainCanvas from "components/MainCanvas";
+import Button from "components/buttons/Button";
+import IconText from "components/IconText";
+import FadeTransition from "components/transitions/FadeTransition";
+import { colors, radius } from "constants/theme";
+
+import IntroEditor from "./introEditor";
+import withCreateSectionIntro from "./withCreateSectionIntro";
+import fragment from "./fragment.graphql";
+
+const StyledMainCanvas = styled(MainCanvas)`
+  padding: 0 2em 1em;
+`;
+
+export const AddIntroButton = styled(Button)`
+  width: 100%;
+  padding: 0.8em 1.8em 0.8em 0.8em;
+  border: 1px solid ${colors.bordersLight};
+`;
+
+export const IntroCanvas = styled.div`
+  padding: 0 1em;
+  border: 1px solid ${colors.bordersLight};
+  border-top: 0;
+  background-color: ${colors.white};
+  border-radius: 0 0 ${radius} ${radius};
+`;
+
+export class UnwrappedSectionIntroduction extends React.Component {
```

Comment:
I guess it needs to be for the `handleCreateSectionIntro` otherwise we are invalidating the tree below for each re-render.

Reply:
Yeah these were both classes just to prevent needing to define the handlers inline. Functional definitely looks better and without knowing whether or not it's a performance hit it's probably better to go with what looks nicer and is easier to follow.